smac/optimizer/random_configuration_chooser.py

```diff
@@ -0,0 +1,47 @@
+from abc import ABC, abstractmethod
+
+__author__ = "Aaron Kimmig"
+__copyright__ = "Copyright 2015, ML4AAD"
+__license__ = "3-clause BSD"
+__maintainer__ = "Aaron Kimmig"
+__email__ = "kimmiga@cs.uni-freiburg.de"
+__version__ = "0.0.1"
+
+
+class RandomConfigurationChooser(ABC):
+    """
+    Abstract base of helper classes to configure interleaving of
+    random configurations in a list of challengers.
+    """
+
+    @abstractmethod
+    def check(self, iteration):
+        """Check if iteration should be followed by a random configuration"""
+        pass
+
+
+class ChooserNoCoolDown(RandomConfigurationChooser):
+
+    def __init__(self, modulus=2):
+        self.modulus = modulus
+
+    def check(self, iteration):
```

Comment:
Just for completeness: Can you set argument types for these methods?

Reply:
If we set modulus < 1 we do random search. We should check for that and indicate that ROAR is the better choice for this.

Reply:
Done setting argument types and adding a warning on modulus <= 1